### PR TITLE
Expose Scaffold.default_local_init_op as a public static method.

### LINKED ISSUE #18242: Make tf.train.Scaffold._default_local_init_op() a public static method?
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yep
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Debian Rolling
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**:  N/A
- **Python version**: 2.7.13
- **Bazel version (if compiling from source)**:  N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Describe the problem

This is a continuation in spirit of Issue #11665, but could we expose tf.train.Scaffold._default_local_init_op() as a **public** static method? This would make it easier in init_fn/init_op/local_init_op to add specific local op initializers to the existing defaults instead of reproducing the logic.

For example:
```
    local_init_op = tf.group(
        tf.train.Scaffold.default_local_init_op(),
        tf.variables_initializer([some_local_var]))
    scaffold = tf.train.Scaffold(
        local_init_op=local_init_op,
        ready_for_local_init_op=tf.report_uninitialized_variables(
            vars_to_load),
        saver=tf.train.Saver(vars_to_load))
```

### PR DESCRIPTION
**Pull request by dthkao**
As discussed offline with @martinwicke, removes the underscore in all repo usages. Fixes #18242

### CODE REVIEW

**Comment by martinwicke** [comment]
Can you run the api_compatibility_test and do what it says when it fails (it will fail)?

**Review by martinwicke — APPROVED** [review]

File: tensorflow/python/training/monitored_session.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def get_or_default(arg_name, collection_key, default_constructor):
     return op
 
   @staticmethod
-  def _default_local_init_op():
+  def default_local_init_op():
     return control_flow_ops.group(
```
Comment (martinwicke) [inline tensorflow/python/training/monitored_session.py]: Please add a docstring.

**Review by martinwicke — CHANGES_REQUESTED** [review]

File: tensorflow/python/training/monitored_session.py (hunk 1 above)
Comment (martinwicke) [reply tensorflow/python/training/monitored_session.py]: Ideally, say which initialization steps this method includes, in which order, so it's possible to use it without looking at the code.

**Comment by dthkao** [comment]
Done.

**Review by martinwicke — CHANGES_REQUESTED** [review]
I believe you need a "Returns: " section in order to not later break the linter.

**Comment by dthkao** [comment]
Sorry about that. Ran lint this time too to make sure.

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
Thanks!

**Pull request merged**